qiita_ware/ebi.py

```diff
@@ -105,14 +104,20 @@ def __init__(self, preprocessed_data_id, action):
 
         ena_ontology = Ontology(convert_to_id('ENA', 'ontology'))
         self.action = action
-        self.preprocessed_data = PreprocessedData(preprocessed_data_id)
-        self.study = Study(self.preprocessed_data.study)
-        self.sample_template = SampleTemplate(self.study.sample_template)
-        self.prep_template = PrepTemplate(self.preprocessed_data.prep_template)
-
-        if self.preprocessed_data.is_submitted_to_ebi and action != 'MODIFY':
-            error_msg = ("Cannot resubmit! Preprocessed data %d has already "
-                         "been submitted to EBI.")
+        self.artifact = Artifact(artifact_id)
+        if not self.artifact.can_be_submitted_to_ebi:
+            error_msg = ("Artifact %d cannot be submitted to EBI"
+                         % self.artifact.id)
+            LogEntry.create('Runtime', error_msg)
+            raise EBISubmissionError(error_msg)
+
+        self.study = self.artifact.study
+        self.sample_template = self.study.sample_template
+        self.prep_template = self.artifact.prep_templates[0]
```

Comment:
Prep templates will be always at least 1. In this specific case, it will be also at most 1. The reason is that if len(prep_templates) > 1 (by design it can't be 0) the property can_be_submitted_to_ebi is False so it will fail on line 108.